l1-contracts-foundry/script/DeployErc20.s.sol

```diff
@@ -0,0 +1,111 @@
+import {ValidatorTimelock} from "contracts/state-transition/ValidatorTimelock.sol";
+
+contract RegisterHyperchainScript is Script {
+    using stdToml for string;
+
+    struct Config {
+        TokenDescription[] tokens;
+        address deployerAddress;
+    }
+
+    struct TokenDescription {
+        address addr;
+        string name;
+        string symbol;
+        uint256 decimals;
+        string implementation;
+        uint256 mint;
+    }
+
+    Config config;
+
+    function run() public {
+        console.log("Deploying ERC20 Tokens");
+
+        initializeConfig();
+        deployTokens();
+    }
+
+    function initializeConfig() internal {
+        // Grab config from output of l1 deployment
+        string memory root = vm.projectRoot();
+        string memory path = string.concat(root, "/script-config/config-deploy-erc20.toml");
+        string memory toml = vm.readFile(path);
+
+        config.deployerAddress = msg.sender;
+
+        // Config file must be parsed key by key, otherwise values returned
+        // are parsed alfabetically and not by key.
+        // https://book.getfoundry.sh/cheatcodes/parse-toml
+        string[] memory tokens = vm.parseTomlKeys(toml, "$.tokens");
+
+        for (uint256 i = 0; i < tokens.length; i++) {
+            TokenDescription memory token;
+            string memory key = string.concat("$.tokens.", tokens[i]);
+            token.name = toml.readString(string.concat(key, ".name"));
+            token.symbol = toml.readString(string.concat(key, ".symbol"));
+            token.decimals = toml.readUint(string.concat(key, ".decimals"));
+            token.implementation = toml.readString(string.concat(key, ".implementation"));
+            token.mint = toml.readUint(string.concat(key, ".mint"));
+
+            config.tokens.push(token);
+        }
+    }
+
+    function deployTokens() internal {
+        for (uint256 i = 0; i < config.tokens.length; i++) {
+            TokenDescription memory token = config.tokens[i];
+            console.log("Deploying token:", token.name);
+            address tokenAddress = deployErc20(
+                token.name,
+                token.symbol,
+                token.decimals,
+                token.implementation,
+                token.mint
+            );
+            console.log("Token deployed at:", tokenAddress);
+            token.addr = tokenAddress;
+        }
+    }
+
+    function deployErc20(
+        string memory name,
+        string memory symbol,
+        uint256 decimals,
+        string memory implementation,
+        uint256 mint
+    ) internal returns (address) {
+        bytes memory args = abi.encode(name, symbol, decimals);
+        bytes memory bytecode = abi.encodePacked(vm.getCode(implementation), args);
+
+        address tokenAddress;
+        vm.broadcast();
+        assembly {
+            tokenAddress := create(0, add(bytecode, 0x20), mload(bytecode))
+        }
```

Comment:
why not reuse create2?

Reply:
I'm migrating the code from hard hat scripts, where normal Create deployment is used. If there's no particular reason this is used, then I'll migrate to Create2

Reply:
just some old stuff